index.php

```diff
@@ -9,6 +9,70 @@
 $query->execute();
 $allInsects = $query->fetchAll();
 
+        if(isset($_POST['submit'])) {
+            $commonName = $_POST['common-name'];
+            $species = $_POST['species'];
+            $dateSpotted = $_POST['date-spotted'];
+            $countrySpotted = $_POST['country-spotted'];
+            $insectSize = $_POST['insect-size'];
+            $addInsect = $db->prepare("INSERT INTO `insect-collection` (`common_name`, `species`, `date_spotted`, 
+            `country_spotted`, `size`, `image_path`) VALUES (:commonName, :species, :dateSpotted, :countrySpotted, :insectSize, :filePath)");
+            $addInsect->bindParam(':commonName', $commonName);
+            $addInsect->bindParam(':species', $species);
+            $addInsect->bindParam(':dateSpotted', $dateSpotted);
+            $addInsect->bindParam(':countrySpotted', $countrySpotted);
+            $addInsect->bindParam(':insectSize', $insectSize);
+            $addInsect->bindParam(':filePath', $filePath);
+            $errorMessages = [];
+            if (!preg_match_all('/^[A-Za-z\s]+$/', $commonName)) {
+                $errorMessages[] = "You did not enter a valid name.";
+            }
+            if (!preg_match_all('/^[A-Za-z\s]+$/', $species)) {
+                $errorMessages[] = "You did not enter a valid species.";
+            }
+            if (!preg_match_all('/^[A-Za-z\s]+$/', $countrySpotted)) {
+                $errorMessages[] = "You did not enter a valid country.";
+            }
+            if (!preg_match_all('/(\\d)+$\s?(mm)?/', $insectSize)) {
+                $errorMessages[] = "You did not enter a valid insect size.";
+            }
+            if (substr($insectSize, -2) !== 'mm') {
+                $insectSize .= ' mm';
+            }
```

Comment:
all these validation things would ideally be in functions and those functions should be unit tested - the reason is because this code technically is not using the db or $_POST - it is simply taking strings, validating them and potentially returning an error message